### PR TITLE
Ledger Nano X Integration: Add Navigation Foundation

### PR DESCRIPTION
**Pull request by jxom**
Fixes PAR-29
Fixes PAR-36
Figma: https://www.figma.com/file/Aw3gB0YvntIBKm1H5iyq0B/Ledger-Nano-X?node-id=88%3A32407

## What changed (plus any additional context for devs)

This PR adds navigation foundations for the hardware wallet pairing screens. Note that the "Pair a hardware wallet" entrypoint may be subject to change, and also the screens are obviously subject to change... Basically, just review the navigation code pls.

## Screen recordings / screenshots

https://www.loom.com/share/9f0bfb7da40446f0a13ff1397391a6ea

### CODE REVIEW

**Comment by jxom** [comment]
@osdnk – I would love your opinion on this :)

**Review by osdnk — APPROVED** [review]
I approve from the navigation perspective!

File: src/components/hardware-wallets/Layout.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import { Box, Inset, Row, Rows } from '@/design-system';
+
+export function Layout({
+  children,
+  header,
+  footer,
+}: {
+  children: React.ReactNode;
```
Comment (osdnk) [inline src/components/hardware-wallets/Layout.tsx:9]: Probably `React.Children`? It's kinda OCP to not enforce children type

File: src/navigation/config.js [hunk 2]
```diff
@@ -150,6 +150,18 @@ export const settingsSheetConfig = {
   }),
 };
 
+export const pairHardwareWalletNavigatorConfig = {
+  options: ({ route: { params = {} } }) => ({
+    ...buildCoolModalConfig({
+      ...params,
+      backgroundOpacity: 1,
+      scrollEnabled: false,
```
Comment (osdnk) [inline src/navigation/config.js:158]: why?

File: src/components/hardware-wallets/Layout.tsx (hunk 1 above)
Comment (jxom) [reply src/components/hardware-wallets/Layout.tsx:9]: yep yep, sounds good

File: src/navigation/config.js (hunk 2 above)
Comment (jxom) [reply src/navigation/config.js:158]: good question, we probs want to enable it actually.

File: src/languages/_english.json [hunk 3]
```diff
@@ -1343,6 +1343,18 @@
       "copied": "Copied \"%{copiedText}\"",
       "invalid_paste": "You can't paste that here"
     },
+    "hardware_wallets": {
+      "intro": {
```
Comment (osdnk) [inline src/languages/_english.json:1347]: I would keep this structure flat, so maybe this mid-level is redundant?

**Review by markdalgleish — APPROVED** [review]

**Pull request merged**